packages/storybook/src/generators/configuration/lib/util-functions.ts

```diff
@@ -630,25 +633,20 @@ export function getTsConfigPath(
 }
 
 export function addBuildStorybookToCacheableOperations(tree: Tree) {
-  updateJson(tree, 'nx.json', (json) => ({
-    ...json,
-    tasksRunnerOptions: {
-      ...(json.tasksRunnerOptions ?? {}),
-      default: {
-        ...(json.tasksRunnerOptions?.default ?? {}),
-        options: {
-          ...(json.tasksRunnerOptions?.default?.options ?? {}),
-          cacheableOperations: Array.from(
-            new Set([
-              ...(json.tasksRunnerOptions?.default?.options
-                ?.cacheableOperations ?? []),
-              'build-storybook',
-            ])
-          ),
-        },
-      },
-    },
-  }));
+  const nxJson = readNxJson(tree);
+
+  if (
+    nxJson.tasksRunnerOptions?.default?.options?.cacheableOperations &&
+    nxJson.tasksRunnerOptions.default.options.cacheableOperations.includes(
```

Comment:
That can error because everything starting in `tasksRunnerOptions` can be `undefined` and it can't be:

```ts
!nxJson?.tasksRunnerOptions?.default?.options?.cacheableOperations?.includes('build-storybook')
```

because we'd then add it when `cacheableOperations` is not defined, which is not what we want. So, we specifically check that `cacheableOperations` is defined and if it is, we then add the target if it's not added already.